.github/workflows/build-test-macos.yml

```diff
@@ -0,0 +1,63 @@
+name: build-and-test-macos
+
+on:
+  push:
+    branches:
+      - master
+
+    paths-ignore:
+      - '**.md'
+      - '.github/**'
+      - '!.github/workflows/build-*'
+      
+
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  build-macos:
+    name: Build-macos
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+    steps:  
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ~1.15.15
+      id: go
+
+    - name: Configure git with longpath enabled (for windows)
+      run: git config --global core.longpaths true
+
+    - name: Check out code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        submodules: 'true'
+
+    - name: Debug go.mod
+      run: cat go.mod
+
+    - name: Cache build output
+      uses: actions/cache@v2
+      with:
+        path: |
+              ~/Library/Caches/go-build
+              ~/go/pkg/mod
+              /Users/runner/Library/Caches/go-build
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Test
+      run: make test
+
```

Comment:
Please  add the 
```
      - name: Upload coverage to Codecov
        uses: codecov/codecov-action@v2
        with:
          verbose: true
```

Reply:
Why do we want to have all of the runners publish test coverage

Reply:
Assuming we retrying a single workflow and due to facts that our test are flakiness, so sometimes we cannot cover all of our tests. Therefore, uploading a test coverage for single workflow is to address this problem.

Reply:
Uploading coverage from all of the OS workflows does not address test flakiness. That being said, it does make sense that we need to run it on all our OS because windows tests would only get run on the windows workflow, and Codecov does [automatic report merging](https://docs.codecov.com/docs/merging-reports), so there's no conflict there. I was under the impression that different code coverage reports would overwrite previous ones on the same commit.